manim/mobject/mobject.py

```diff
@@ -237,7 +235,7 @@ def construct(self):
             cls.__init__ = cls._original__init__
 
     @property
-    def animate(self: T) -> _AnimationBuilder | T:
+    def animate(self) -> _AnimationBuilder | Self:
```

Comment:
What was the reason to have the union return type in the first place? Is it because `_AnimationBuilder` acts as some kind of proxy object? If so, there's probably a better way to type this, but let's keep this for another PR when this module will be tackled as part of the typing progress

Reply:
> s it because _AnimationBuilder acts as some kind of proxy object?

That's the gist of it!